### PR TITLE
[WIP] Widget modal

### PR DESCRIPTION
**Pull request by bumi**

### CODE REVIEW

File: src/components/modal/index.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    return () => {
+      document.body.style.overflow = "visible";
+      document.removeEventListener("keydown", onKeyDown);
+    };
+  }, []);
+
+  return (
+    <Backdrop onClick={onRequestClose}>
+      <ModalContainer>
+        <Button buttonClick={onRequestClose} btnText="Close modal" />
```
Comment (bumi) [inline src/components/modal/index.js:31]: As we have to render all the views in the modal: that means we load all the components from this modal component?

Comment (reznord) [reply src/components/modal/index.js:31]: Yes, we will render different views as a stepper flow.

File: src/views/widget/index.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+import { useState } from "preact/hooks";
+// import { WidgetWrapper, Title } from "./styled";
+
+import Modal from "../../components/modal";
+import Button from "../../components/button";
+
+const Widget = () => {
+  const [isModalOpen, setModalIsOpen] = useState(false);
+
+  const toggleModalVisibility = () => {
```
Comment (bumi) [inline src/views/widget/index.js:11]: How do we make this available from the outside? 

I want users to be able to have their custom html to open the modal. 

e.g. I have a link that opens the widget

```html
<a href="#" onclick="Widget.open()">open</a>
```

Comment (reznord) [reply src/views/widget/index.js:11]: I was going with the initial flow of adding the button to the HTML via the classname. So, I am thinking of completing that first.
```HTML
...
<div data-widget-id=lighitning-payments />
...
```

I will check on how to expose the `Widege.open()` from the HTML parts.

**Pull request merged**